### PR TITLE
fix: fix depends builds on FreeBSD hosts

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

Despite builders for BSD-based platforms being backported as early as [dash#5362](https://github.com/dashpay/dash/pull/5362), they didn't work on account of using a `curl` flag, `--retry`, that was lopped off other builders in [dash#3003](https://github.com/dashpay/dash/pull/3003) as a way to mitigate aberrant Travis CI-specific behaviour.

As the variable `$(DOWNLOAD_RETRIES)` was removed as part of that mitigation, the introduction of builders that rely on this variable caused failures as the flag was accompanied by nothing. As our CI host isn't based on FreeBSD, this went unnoticed. When deciding between extending the existing patch and reverting it, reverting it proved to be more attractive on account of us no longer using Travis CI and the revert bringing us closer to upstream.

Additionally, the `std::unary_function` patch that was introduced in [dash#5610](https://github.com/dashpay/dash/pull/5610) proves to be necessary on BSD-based platforms as well (had to be extended for and tested on GhostBSD, based on FreeBSD 13.2-STABLE, Clang 16). Instead of conditionally patching based on platform and compiler, a more reliable patch would be to downgrade the C++ version used to build Boost at the last version to have `std::unary_function`, which would be C++11, albeit deprecated ([source](https://en.cppreference.com/w/cpp/utility/functional/unary_function)).

Though it's likely this route wasn't taken originally due to compiler errors that happened despite downgrading to C++11. These errors were due to the compiler objecting to `std::unary_function` usage in Boost headers, despite the backport of [bitcoin#25436](https://github.com/bitcoin/bitcoin/pull/25436), which should've solved this problem. The reason the errors were still persisting is because the necessary flag, `-DBOOST_NO_CXX98_FUNCTION_BASE`, was only applied if `--enable-suppress-external-warnings` was set. CI didn't catch this, as the flag is always set, to keep log lengths manageable. This has been rectified.

All changes combined, one should be able to build non-Qt Dash binaries using `depends` though building the Qt client from `depends`  unfortunately remains a problem, even upstream ([source](https://github.com/bitcoin/bitcoin/pull/23955#issuecomment-1039300427), [source](https://github.com/bitcoin/bitcoin/pull/23948#issue-1092284497)).

## Demo

**Based on bbc99571f36ecfd817dc33b883c5e6f120240270**

Built using:
* **`depends` flags:** `NO_QT=1 ALLOW_HOST_PACKAGES=1 HOST=x86_64-unknown-freebsd13.2`
* **`configure` flags :** `--prefix=$(pwd)/depends/x86_64-unknown-freebsd13.2 --enable-debug --enable-suppress-external-warnings --with-gui` 
* Qt installed using `pkg` (`qt5`)

Dash-Qt running on GhostBSD

## Checklist:

- [x] I have performed a self-review of my own code **(note: N/A)**
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests **(note: N/A)**
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
Compiled with no issues on macos 14.5, Apple clang version 15.0.0 (clang-1500.3.9.4)

light ACK bbc99571f36ecfd817dc33b883c5e6f120240270

File: depends/packages/boost.mk [hunk 1]
```diff
@@ -27,7 +27,7 @@ else
 $(package)_toolset_$(host_os)=gcc
 endif
 $(package)_config_libraries=filesystem,test
-$(package)_cxxflags=-std=c++17
+$(package)_cxxflags=-std=c++11
```
Comment (knst) [inline depends/packages/boost.mk:30]: could you add TODO here so far as it is not permanent change?

Comment (kwvg) [reply depends/packages/boost.mk:30]: Mentioned it in https://github.com/dashpay/dash-issues/issues/53

**Review by PastaPastaPasta — APPROVED** [review]
utACK bbc99571f36ecfd817dc33b883c5e6f120240270

File: depends/packages/boost.mk (hunk 1 above)
Comment (knst) [reply depends/packages/boost.mk:30]: I meant TODO right here, so, next release we can re-evaluate it if it is still necessary or can be removed, but I guess you should not forget it yourself :D

**Review by knst — APPROVED** [review]
utACK bbc99571f36ecfd817dc33b883c5e6f120240270

**Pull request merged**